bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/module/provider/i18n/ModuleTypeI18nService.java

```diff
@@ -32,6 +35,8 @@
      * @param bundle the bundle containing the localization files
      * @return the localized ModuleType
      */
-    ModuleType getModuleTypePerLocale(ModuleType defModuleType, Locale locale, Bundle bundle);
+    @Nullable
+    ModuleType getModuleTypePerLocale(@Nullable ModuleType defModuleType, @Nullable Locale locale,
```

Comment:
Are you sure the `Bundle` should be annotated with `@Nullable`? Because if we do not have a bundle we will not get a translation. Thus I think is does not make sense to call such a method in that case. We e.g. did not annotated it with `@Nullable` in the `ConfigI18nLocalizationService` or `ProfileI18nUtil`.

Reply:
Looks like I used `@Nullable` because the `bundle` in `org.openhab.core.automation.internal.provider.ModuleTypeResourceBundleProvider.getPerLocale` can be `null` :

```java
        String uid = defModuleType.getUID();
        Bundle bundle = getBundle(uid);
```

We can also let `getPerLocale` returns `null` when it can't find the bundle.

Reply:
I tried to look into this. At the first glance I second your suggestion to return `null` if the `bundle` is `null`.

At the second view I would tend to throw away the `getBundle()` method it somehow looks strange to me. In other bundles we use the `FrameworkUtil.getBundle()` method in such cases. But to be honest I do not now if it will work here.

Reply:
Until I fully understand what is going here I've gone for the conservative approach by adding a null check and not throwing the method away. I think the idea is that these ModuleTypes can be provided by different bundles.